cloud/deployment/deployment_test.go

```diff
@@ -250,8 +269,25 @@ var (
 )
 
 func (s *Suite) SetupTest() {
+	// init mocks
+	mockPlatformCoreClient = new(astroplatformcore_mocks.ClientWithResponsesInterface)
+	mockCoreClient = new(astrocore_mocks.ClientWithResponsesInterface)
+
+	// init responses object
+	MockResponseInit()
+}
+
+func (s *Suite) TearDownSubTest() {
```

Comment:
This fixes the state leak issue between unit tests from https://github.com/astronomer/astro-cli/issues/1698, I had to do this bit here because otherwise, the existing unit tests were making it almost impossible to write new unit tests, and it took me 3 hours to write a single test in this file 🤷